src/local-development/code-pull-deploy.ts

```diff
@@ -36,6 +38,9 @@ export async function pullComponentCode({
 	destinationPath: vscode.Uri;
 }): Promise<void> {
 	log('debug', `Pull ${appComponentType} ${remoteComponentName}: code ${codeType}`);
+
+	sendTelemetry('pull_component_code', { appComponentType, remoteComponentName });
+
```

Comment:
On other places, you have telemetry AFTER real action. Here is is BEFORE. For consistency choose the one variant and folow it everywhere.
If you will call if BEFORE, it will be logged on every actions (includes failed actions). If you will call it AFTER, then the event not logged on function error.

Reply:
moved before action - we are focusing on user interactions, not error handling